app/models/repo.rb

```diff
@@ -80,23 +85,48 @@ def pr_branch_names
     pr_branches.collect(&:name)
   end
 
-  def stale_pr_branches
-    pr_branch_names - github_branch_names
+  # @param expected [Array<Hash>] The desired state of the PR branches.
+  #   Caller should pass an Array of Hashes that contain the PR's number,
+  #   html_url, and merge_target.
+  def synchronize_pr_branches(expected)
+    raise "repo cannot not have PR branches" unless can_have_prs?
+
+    git_fetch # TODO: Let's get rid of this!
+
+    transaction do
```

Comment:
Do we need to lock the branch records for the duration of the transaction?  I'm thinking other workers could come along and change something while we're in a transaction.